examples/fluids/qfunctions/densitycurrent.h

```diff
@@ -108,25 +121,23 @@ CEED_QFUNCTION_HELPER int Exact_DC(CeedInt dim, CeedScalar time,
 
   // -- Exner pressure, hydrostatic balance
   const CeedScalar Pi = 1. + g*g*(exp(-N*N*z/g) - 1.) / (cp*theta0*N*N);
-  // -- Density
-
-  const CeedScalar rho = P0 * pow(Pi, cv/Rd) / (Rd*theta);
 
   // Initial Conditions
-  q[0] = rho;
-  q[1] = 0.0;
-  q[2] = 0.0;
-  q[3] = 0.0;
-  q[4] = rho * (cv*theta*Pi + g*z);
-
-  return 0;
+  CeedScalar Y[5] = {0.};
+  Y[0] = P0 * pow(Pi, cv/Rd);
+  Y[1] = 0.0;
+  Y[2] = 0.0;
+  Y[3] = 0.0;
+  Y[4] = theta;
+
+  return StateFromY(gas, Y, X);
 }
 
 // *****************************************************************************
-// This QFunction sets the initial conditions for density current
+// These QFunctions set the initial conditions for density current
 // *****************************************************************************
-CEED_QFUNCTION(ICsDC)(void *ctx, CeedInt Q,
-                      const CeedScalar *const *in, CeedScalar *const *out) {
+CEED_QFUNCTION(ICsDC_Prim)(void *ctx, CeedInt Q,
```

Comment:
Could also use a `DensityCurrentContext->newtonian_ctx.primitive` to make this a single ICs.